.github/workflows/create_release.yml

```diff
@@ -254,11 +278,37 @@ jobs:
           verbose: true     
           print-hash: true  
 
-  test_source_dist:
-    name: test source distribution
-    needs: [publish_preview_build_to_pypi_weekly]
-    if: github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'run release CIs')
-    uses: ./.github/workflows/preview_source_dist_test.yml
-    with:
-      os: "macos"
+  publish_release_build_to_pypi:
+    name: Publish release build to pypi
```

Comment:
I wonder if we should add this check: ensure that the package version is equal to the release-* branch version

Reply:
I would add the checks to "check_for_publish_release_build_to_pypi:"